src/provider/bn256_grumpkin.rs

```diff
@@ -10,13 +10,11 @@ use crate::{
 };
 use digest::{ExtendableOutput, Update};
 use ff::{FromUniformBytes, PrimeField};
+use group::{cofactor::CofactorCurveAffine, Curve, Group as AnotherGroup, GroupEncoding};
 use num_bigint::BigInt;
 use num_traits::Num;
-use pasta_curves::{
```

Comment:
Should we make a similar change in src/provider/pasta.rs?

Reply:
I do not expect the version of the `::group` traits re-exported in the `::pasta_curves` crate to ever be out of sync w.r.t the traits implemented by the `::pasta_curves` structs. 
The point of the change you’re commenting on is to reflect that the `halo2curves` crate, while using crates from `::group`, is not particularly bound (notably in their versions) to those re-exported by the `pasta_curves` crates, with which it does not have a relationship.

Reply:
Yes, that makes sense, thanks!